metis-indexing/src/main/java/eu/europeana/indexing/utils/SetUtils.java

```diff
@@ -19,12 +17,48 @@ private SetUtils() {}
 
   /**
    * <p>
-   * This method generates all possible combinations of elements of different sets. Each combination
-   * consists of zero or one element of each set. As such, the number of combinations of sets s1,
-   * s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1) * ... * (|sn| + 1),
-   * where |s| denotes the cardinality (size) of set s. However, depending on the choice of the
-   * types and the concatenator, not all of these may be unique and the result set may contain fewer
-   * items.
+   * This method generates all possible combinations of elements of different sets.
+   * </p>
+   * <p>
+   * Each combination consists of zero or one element of each set. As such, the number of
+   * combinations of sets s1, s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1)
```

Comment:
I think it's right. From each set you take zero or one elements. So for set s1 there are (|s1| + 1) options. 

Like I wrote somewhere else:
As an example: suppose the web resource has values a1 and a2 for facet a, and b1 for facet b.
Then this method will return the following six possible combinations:
   * [] (the empty value)
   * [a1]
   * [a2]
   * [b1]
   * [a1 | b1]
   * [a2 | b1]
But it will not return [a1 | a2] or [a1 | a2 | b1]

Reply:
I see, I disregarded the "emtpyCombination", since I interpreted wrong by its name therefore I've made the other comment.
It makes sense now since the "emtpyCombination" reflects that combination with one item.